File: backend/app/models/property.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    ForeignKey,
    Enum,
    UniqueConstraint,
)

from sqlalchemy.orm import relationship

from app.database.base_class import Base
from app.enums import PropertyType


class Property(Base):

    __tablename__ = "properties"

    __table_args__ = (
        UniqueConstraint(
            "organization_id",
            "name",
            name="uq_property_organization_name",
        ),
    )

    id = Column(
        Integer,
        primary_key=True,
        index=True,
    )

    name = Column(
        String(100),
        nullable=False,
    )

    property_type = Column(
        Enum(PropertyType),
        nullable=False,
    )

    organization_id = Column(
        Integer,
        ForeignKey(
            "organizations.id",
            ondelete="CASCADE",
        ),
        nullable=False,
    )

    organization = relationship(
        "Organization",
        back_populates="properties",
    )

    sections = relationship(
        "Section",
        back_populates="property",
        cascade="all, delete",
    )

    units = relationship(
        "Unit",
        back_populates="property",
        cascade="all, delete",
    )

    address = Column(
        String,
        nullable=True,
    )

    city = Column(
        String,
        nullable=True,
    )

    state = Column(
        String,
        nullable=True,
    )

    country = Column(
        String,
        nullable=True,
    )

    pincode = Column(
        String,
        nullable=True,
    )

File: frontend/safecolony_app/lib/features/admin/property/models/property.dart
class Property {
  final int id;
  final String name;
  final String propertyType;
  final bool hasMultipleSections;
  final int organizationId;
  final String address;
  final String city;
  final String state;
  final String country;
  final String pincode;

  Property({
    required this.id,
    required this.name,
    required this.propertyType,
    required this.hasMultipleSections,
    required this.organizationId,
    required this.address,
    required this.city,
    required this.state,
    required this.country,
    required this.pincode,
  });

  factory Property.fromJson(Map<String, dynamic> json) {
    return Property(
      id: json["id"],
      name: json["name"],
      propertyType: json["property_type"],
      hasMultipleSections: json["has_multiple_sections"] ?? false,
      organizationId: json["organization_id"],
      address: json["address"],
      city: json["city"],
      state: json["state"],
      country: json["country"],
      pincode: json["pincode"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "name": name,
      "property_type": propertyType,
      "has_multiple_sections": hasMultipleSections,
      "organization_id": organizationId,
      "address": address,
      "city": city,
      "state": state,
      "country": country,
      "pincode": pincode,
    };
  }
}